trie/utils/binaries.py

```diff
@@ -0,0 +1,66 @@
+from trie.constants import (
+    TWO_BITS,
+    PREFIX_00,
+    PREFIX_100000,
+)
+
+
+def safe_ord(value):
+    if isinstance(value, int):
+        return value
+    else:
+        return ord(value)
+
+
+def decode_from_bin(input_bin):
```

Comment:
I just toyed with an alternate implementation of this.

```python
from eth_utils import (
    apply_to_return_value,
)
from cytoolz import (
    partition_all,
)


@apply_to_return_value(bytes)
def alt_decode_from_bin(input_bin):
    for chunk in partition_all(8, input_bin):
        yield sum(
            2**exp * bit
            for exp, bit
            in enumerate(reversed(chunk))
        )
```

Some *very* rudimentary benchmarking suggests it's about 30% faster than the current implementation.